src/DictVectors/stochasticstyle.jl

```diff
@@ -58,17 +68,21 @@ stochastically projected to either zero or `threshold`.
 
 See also [`StochasticStyle`](@ref).
 """
-struct IsStochasticWithThreshold <: StochasticStyle{Float64}
-    threshold::Float64
+struct IsStochasticWithThreshold{T<:AbstractFloat} <: StochasticStyle{T}
+    threshold::T
 end
+IsStochasticWithThreshold(t=1.0) = IsStochasticWithThreshold{Float64}(Float64(t))
 
 """
     IsDynamicSemistochastic(rel_threshold=1, abs_threshold=Inf, proj_threshold=1) <: StochasticStyle
 
 Similar to [`IsStochasticWithThreshold`](@ref), but does exact spawning when the number of
 walkers in a configuration is high.
```

Comment:
Say what it does rather than ask the reader to read the docstring of `IsStochasticWithThreshold` first! 
```suggestion
QMC propagation with non-integer walker numbers and reduced noise. All possible spawns are performed 
deterministically when number of walkers in a configuration is high. Vector compression with threshold 
`proj_threshold` is applied after spawning and diagonal death steps.
```